[fix] Исправлена логика login:

xDefDefDefx opened the pull request:
AuthenticationRestController,
UserDetailsServiceImpl,
UserTokenDto

xDefDefDefx committed:
[fix] Исправлена логика login:
AuthenticationRestController,
UserDetailsServiceImpl,
UserTokenDto

NikitaNesterenko commented on security/src/main/java/jm/stockx/UserDetailsServiceImpl.java:
```diff
@@ -36,4 +44,18 @@ public UserDetails loadUserByUsername(String email) throws UsernameNotFoundExcep
         return user;
     }
+
+    public Authentication authenticateUser(@RequestBody UserLoginDto loginUser) {
```
зачем аннотация реквест боди?

NikitaNesterenko commented on security/src/main/java/jm/stockx/UserDetailsServiceImpl.java:
```diff
@@ -36,4 +44,18 @@ public UserDetails loadUserByUsername(String email) throws UsernameNotFoundExcep
+
+            return authentication;
+        } catch (AuthenticationException e) {
+            e.getMessage();
```
убрать вывод месседжа, необходимо чтоб эдвайс перехватывал исключение и отправлял сообщение

NikitaNesterenko commented on webapp/src/main/java/jm/stockx/rest_controller/AuthenticationRestController.java:
```diff
@@ -21,49 +23,33 @@
-            } catch (UserNotFoundException e) {
-                e.printStackTrace();
-            }
+            User foundUser = userService.getUserByEmail(loginUser.getEmail());
```
зачем тащить всего юзера? выше в сервисе есть данные с секьюрити

NikitaNesterenko commented on webapp/src/main/java/jm/stockx/rest_controller/AuthenticationRestController.java:
```diff
@@ -21,49 +23,33 @@
-            } catch (UserNotFoundException e) {
-                e.printStackTrace();
-            }
+            UserTokenDto userTokenForResponse = new UserTokenDto(foundUser, createdToken);
```
передавать необходимые данные в конструктор
переделать конструктор

NikitaNesterenko commented on webapp/src/main/java/jm/stockx/rest_controller/AuthenticationRestController.java:
```diff
@@ -21,49 +23,33 @@
+        } catch (UserNotFoundException e) {
+            e.getMessage();
         }
         return null;
```
убрать налл

xDefDefDefx committed:
[fix] Исправлена логика login:
Основная логика аутентификации вынесена в UserDetailsServiceImpl;
AuthenticationRestController = "тонкий" контроллер;
UserTokenDto переиспользует данные Authentication.

xDefDefDefx committed:
[test] Добавлен тест класс для:
UserDetailsServiceImpl
